### ISSUE TITLE
Interest ranking - What's your Jam?

### ISSUE THREAD

**Issue by marleighnorton** [neinteractiveliterature/intercode#2493]
So the idea here is to give players a way to record what they're interested in, ideally replacing all the notes players take outside the system, e.g. in spreadsheets.

Goals
- Players can sort games by interest
- Players can filter out games they don't care to play
- New games can easily be found
- This should should be private to each player.

Current thinking
- 5 point Jam scale of Not My Jam through Totally My Jam (language pending), as well as TBD.  All games are initially set as TBD.
- Filters added to Events page to filter out various values.  Most likely used are hiding the Not My Jam games and showing only the TBD games, as those are the ones players still need to think about.
- Sort Events page based on Jam.

**Comment by jhartwin** [comment]
I love this idea, but a 5 point scale seems far too granular to me. I'd want "I definitely don't want to play this" and "I definitely want to play this", and could see a use for (but would probably not myself use) "I don't know if I want to play this, but I want to be able to include it in a filter." I'm certainly not objecting to a more granular scale, and if we do end up with a 5 point scale, I'll just use the two extreme ends and ignore the middle and that will be fine. But I'm curious what your intended use case is for 5 points.

**Comment by dkapell** [comment]
A small note here:  These ratings probably need to not be visible to the
GMs, although there's a chance that con staff will want to see them to help
judge how they built the slate of games.

**Comment by marleighnorton** [comment]
5 points:  Honestly, just because it's what I used when I built my own spreadsheet last time.  :)
More than two was definitely helpful to me.  I was sort of
1. Nope.  Just nope.
2. Eh, if I don't have anything else going on then
3. Has some good things going for it
4. I'd like to play, but there's some others I'm more excited about.
5. If I can only get into a few games this entire weekend, this is one of them.

Privacy:  We definitely do not show this to GMs.  I'd argue we don't show it to staff either, that it remains private to the user.  We don't know why they don't want to play a game.  Maybe it's great, they just have played it before, and we don't want people not to trust the feature because they might be influencing games in some unexpected way.  I feel like signups/waitlists give a lot of that feedback already.

**Comment by dkapell** [comment]
If we do share with con staff, it would be anonymized:  "7 people marked
this game as 1, 5 as 2, 15 as 3, and so on" . And, to be clear, I'm not
sure it's a good idea, i just wanted to capture this thought

**Comment by nbudin** [comment]
Re the privacy issue, I'm uncomfortable even being able to see this stuff as a site admin, and even in aggregate.  I'd like to be able to make the guarantee to attendees that their preferences are theirs alone and nobody else can see them (modulo, like, going to look in the database, which we promise not to).

**Comment by dkapell** [comment]
That's a reasonable stance, lets plan for that and revisit if needed.
--dave

**Comment by marleighnorton** [comment]
Prior art.  So rating systems are the things I've been looking into, which is a bit of a fallacy.  We're not really asking people to rate the game per se, but rather let them take notes in a lightweight way.  One of the problems with cribbing off rating systems is whether people actually understand what we're trying to help them do.  We're not asking them to rate the game, but help them remember which games they're interested in playing.  But I'm not seeing a lot of prior art on that, so here's what I'm seeing on ranking systems.

Netflix, Youtube
Used to have a multistar rating system, but switched to thumbs up/thumbs down.  Both thumbs up and thumbs down are shown until one is clicked, at which point that icon is filled in and is the only one shown.  To clear, tap the icon again.
So only three states (interested, not interested, clear) which would certainly be simpler.  Enough granularity for us, though?

Amazon, Lyft, Rotten tomatoes, Yelp
5 star system.  Interaction involves a subdialogue where you create a review, and you either clear the score by deleting the review or can't change it once entered.  That doesn't work for us, but stars could be nice and simple.  Just tap the star you want to set it to.  Need some way to clear stars, though.  I'm not seeing an established design pattern on the site to clear a thing.  Anyone know of one?

Facebook, Net Promoter Score
A set of emoticons/facial expressions.  Might get away from some of the value judgements depending on which ones we use.  Heart, thumbs up, questioning face, sleepy face?

Will synthesize this into proposals in the next comment.

**Comment by marleighnorton** [comment]
Proposal 1: Interested, Not Interested, TBD

On the theory that design is hard and we'll always do things wrong the first time, we could just start with something simple, see how it's received, and then iterate on it as needed.

Events are TBD by default.  You can set them to Interested, Not interested, and back to TBD.  Possibly we'll crib off the Netflix interaction, which is quite elegant.  We'll have to make it clear this is separate from signups, it's not affecting number of runs or anything, it's just your personal notes.  Which I'm concerned some folks will not get, even if we say it bunches.

Proposal 2:  Multistate Nonsense.  Like, literally jam.

Ok, this is exceptionally silly, but hear me out.

We have some number of states which correspond to flavors of jam.  Let's say Strawberry Jam, Grape Jelly, Lemon Marmelade, Apricot Preserves, and Jalapeño Relish.  Which are all conveniently of unique colors and shapes, oho!  As always, games start as TBD.  You assign a jam to each game, or can clear it back to TBD.

My theory is that with flavors, people have a value system that they bring to it, but it's different between people.  Privacy is guaranteed on our end, because we literally don't know how the player is using them.  Like for me, it's Strawberry Jam, Apricot Preserves, Grape Jelly, Lemon Marmelade, then Jalapeño Relish.  And I don't have to worry about forgetting that, because I can rederive it just by looking.  I love Strawberry Jam.  I don't like sour/spicy things.  Grapes vs Apricots are kind of a toss up for me, but eh, definitely in the not my first choice but entirely ok vicinity.

So yes, thoughts?

**Comment by nbudin** [comment]
My 2 cents: I like proposal 1 a lot.  It’s simple, doesn’t require a lot of explanation, and clearly seems like something people would use.  We could easily turn on sorting by it by default in the list of events view.

I see where you’re going with proposal 2, and I like the idea of everyone having their own ranking system that we couldn’t possibly understand even if we tried from a privacy POV.  But I worry that the weirdness of it would discourage people from using it.  We also couldn’t possibly sort by it, so we would have to make people go through a few clicks or do bookmarking to get to the event list they want to see.

So basically, I lean towards proposal 1.

**Comment by marleighnorton** [comment]
Yeah, after thinking about it some more, the other problem with proposal 2 is we can't sort in order, since we don't know what they mean by Lemon Marmelade.  Which isn't the worse, you can still filter, but it's not quite as handy.

For proposal 1, just the three states you think?  Keep it simple for now?  If we did, any chance we could get it in for Intercon, or don't want to risk it?

**Comment by nbudin** [comment]
Yeah, I think just the three states.  I’m definitely gonna try for getting it in ASAP.  This would be a big help during the release of the schedule and there’s nothing I can see that’s super urgent on the project board.

**Comment by jhartwin** [comment]
I am pretty nervous about people thinking this information is going to be publicly available, or available to GMs and/or con staff. I'm totally fine with getting this on the site before signups, but I'd ask that we be very explicit and obvious that these ratings are private and for personal use and no-one other than the user will ever have access to them.

**Comment by marleighnorton** [comment]
@jhartwin Do you have thoughts about language we should use to make that more clear?

I was also considering Show/Hide, but that implies hiding things by default, which is another change.  Not necessarily a bad one, but something to remember.

**Comment by jhartwin** [comment]
I mean, I think we can use the language I drafted above: "These rating are intended for personal use and will always be private; no-one other than the user will ever have access to them."

(If we do go with show/hide, I'd suggest that we show things by default and make an obvious "hide this game from my view" button, rather than hiding by default which I think would just confuse people.)

**Comment by marleighnorton** [comment]
Ok, we can do that.  I think the Interested/Not interested/TBD language is clearest, so I advocate for that for now.

(show/hide: I just meant that the default on the list of events is to show everything.  But if someone has explicitly marked something as hidden, we hide it by default.  As opposed to interested/not interested, which I think we still show everything by default and let people filter out the not interested ones.)
